### PR TITLE
CDD-2033: online pull part of retry

### PR DESCRIPTION
**Pull request by paulblakehmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/notification/services/customsNotificationServices.scala [hunk 1]
```diff
@@ -89,22 +90,32 @@ class CustomsNotificationRetryService @Inject()(logger: NotificationLogger,
...
+      }.flatMap { result =>
           if (result) {
             notificationWorkItemRepo.setCompletedStatus(workItem.id, Succeeded)
-            logger.info(s"push succeeded for workItemId ${workItem.id.stringify}")
+            logMsgBuilder.append(" succeeded").append(s" for workItemId ${workItem.id.stringify}")
+            logger.info(logMsgBuilder.toString())
           } else {
             notificationWorkItemRepo.setCompletedStatus(workItem.id, PermanentlyFailed)
-            logger.error(s"push failed for workItemId ${workItem.id.stringify}")
+            logMsgBuilder.append(" failed").append(s" for workItemId ${workItem.id.stringify}")
```
Comment (rashadmughal) [inline app/uk/gov/hmrc/customs/notification/services/customsNotificationServices.scala]: call append only once.

File: app/uk/gov/hmrc/customs/notification/services/customsNotificationServices.scala [hunk 2]
```diff
@@ -89,22 +90,32 @@ class CustomsNotificationRetryService @Inject()(logger: NotificationLogger,
...
+            logger.info(logMsgBuilder.toString())
           }
           Future.successful(true)
         }.recover {
```
[the remark this answers is not in the document]
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/notification/services/customsNotificationServices.scala]: Fixed

File: test/unit/services/PullClientNotificationRetryServiceSpec.scala [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      val result = await(service.send(NotificationWorkItem1))
+
+      result shouldBe false
+      PassByNameVerifier(mockLogger, "error")
+        .withByNameParam("[conversationId=eaca01f9-ec3b-4ede-b263-61b626dde231][clientSubscriptionId=eaca01f9-ec3b-4ede-b263-61b626dde232]Failed to pass the notification to PULL service")
+        .withByNameParamMatcher(any[RuntimeException])
+        .verify()
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (paulblakehmrc) [reply test/unit/services/PullClientNotificationRetryServiceSpec.scala]: Fixed

File: app/uk/gov/hmrc/customs/notification/services/customsNotificationServices.scala [hunk 4]
```diff
@@ -89,22 +90,32 @@ class CustomsNotificationRetryService @Inject()(logger: NotificationLogger,
...
+          pullClientNotificationRetryService.send(notificationWorkItem)
+        } else {
+          logMsgBuilder.append("push")
+          pushClientNotificationRetryService.send(apiSubscriptionFields, workItem.item)
+        }
+      }.flatMap { result =>
           if (result) {
             notificationWorkItemRepo.setCompletedStatus(workItem.id, Succeeded)
-            logger.info(s"push succeeded for workItemId ${workItem.id.stringify}")
+            logMsgBuilder.append(" succeeded").append(s" for workItemId ${workItem.id.stringify}")
```
[the remark this answers is not in the document]
Comment (paulblakehmrc) [reply app/uk/gov/hmrc/customs/notification/services/customsNotificationServices.scala]: Fixed

**Pull request merged**